File: lib-android/src/main/java/com/pranav/lib_android/code/disassembler/ClassFileDisassembler.java
package com.pranav.lib_android.code.disassembler;

import kotlin.io.FilesKt;

import org.eclipse.jdt.internal.core.util.Disassembler;

import java.io.File;
import java.io.IOException;

public class ClassFileDisassembler {

    final byte[] classFileBytes;

    public ClassFileDisassembler(String classFile) throws IOException {
        classFileBytes = FilesKt.readBytes(new File(classFile));
    }

    public String disassemble() throws Throwable {
        return new Disassembler().disassemble(classFileBytes, System.lineSeparator());
    }
}

File: lib-android/src/main/java/com/pranav/lib_android/code/disassembler/ClassFileDisassembler.kt
package com.pranav.lib_android.code.disassembler

import java.io.File

import org.eclipse.jdt.internal.core.util.Disassembler

class ClassFileDisassembler(classFile: String) {

  val classFileBytes: ByteArray

  init {
    classFileBytes = File(classFile).readBytes()
  }
	
	fun disassemble(): String {
		return Disassembler()
		    .disassemble(classFileBytes, System.lineSeparator())
	}
}
